x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/InferTrainedModelDeploymentAction.java

```diff
@@ -92,12 +92,41 @@ public static Request.Builder parseRequest(String deploymentId, XContentParser p
         private final InferenceConfigUpdate update;
         private final TimeValue inferenceTimeout;
         private boolean skipQueue = false;
+        // textInput added for the semantic search endpoint
+        // which accepts a query string rather than a document
+        private final String textInput;
```

Comment:
I don't think this is strictly necessary. Is this to cover the scenario when we don't know text field name for the model? Or is this just some internal sugar?

Reply:
I see that this is indeed to cover the scenario when the input field is unknown and the input is just text. 

Could we just reuse [NlpInferenceInput.java](https://github.com/elastic/elasticsearch/pull/90450/files/77bab64a66d50a32fa33d18b87b30667dcfbc6f0#diff-103e3d3c45b855a9b9c95f83c74d1dbb52be66653b9e3ace9ed49bed06d693b8) up here?

Reply:
Yes it is for the scenario where the input field is unknown. It is not possible to rationalise this as the inference processor is the opposite scenario: it has a document but no idea which specific field in the document to use. 

 `NlpInferenceInput` is designed to abstract away that detail but if I used it here I would have to pull it apart to perform BWC wire serialisation.